downloader/src/main/kotlin/Downloader.kt

```diff
@@ -77,7 +80,32 @@ object Downloader {
 
         val exception = DownloadException("Download failed for '${pkg.id.toCoordinates()}'.")
 
-        // Try downloading from VCS.
+        return config.sourcesOrigin.asSequence().mapNotNull { origin ->
```

Comment:
I don't insist on it, but still believe `distinct()` would be of some value here. Just as a reminder, in case you forgot it.